src/com/itacademy/eshop/services/ShopService.java

```diff
@@ -13,8 +14,8 @@ public Eshop createShop() {
         /** Should create new e-shop object and initialize it with some products.
          * Should return e-shop object.
          */
-       ArrayList<Product> products = createProducts();
-        throw new UnsupportedOperationException("createShop() is not implemented yet");
+        ArrayList<Product> products = createProducts();
+        return new Eshop("name", products);
     }
 
     private ArrayList<Product> createProducts() {
```

Comment:
You could opt out for a generic using List interface to have a big more freedom for the types of lists you can use. You have already learned about interfaces